http/src/main/java/io/micronaut/http/util/HttpHeadersUtil.java

```diff
@@ -135,4 +140,67 @@ public static String splitAcceptHeader(@NonNull String text) {
         }
         return text;
     }
+
+    /**
+     * Resolve the {@link Charset} to use for request identified by the Content-Type HTTP Header value and the Accept-Charset HTTP Header value.
+     *
+     * @param contentTypeHeaderValue Content-Type HTTP Header Value
+     * @param acceptCharsetHeaderValue Accept-Charset HTTP Header Value
+     * @return A {@link Charset}
+     * @since 4.8.8
+     */
+    @NonNull
+    public static Charset parseCharacterEncoding(@Nullable String contentTypeHeaderValue, @Nullable String acceptCharsetHeaderValue) {
+        MediaType contentType = contentTypeHeaderValue == null ? null : MediaType.of(contentTypeHeaderValue);
+        Charset charset = acceptCharsetHeaderValue != null ? parseAcceptCharset(acceptCharsetHeaderValue) : StandardCharsets.UTF_8;
+        return parseCharacterEncoding(contentType, charset);
+    }
+
+    /**
+     * Resolve the {@link Charset} to use for the request.
+     *
+     * @param contentType ContenType
+     * @param acceptCharset Accept Charset
+     * @return An {@link Optional} of {@link Charset}
+     * @since 4.8.8
+     */
+    @NonNull
+    public static Charset parseCharacterEncoding(@Nullable MediaType contentType,
+                                                 @NonNull Charset acceptCharset) {
+        try {
+
+            if (contentType != null) {
+                String charset = contentType.getParametersMap().get(MediaType.CHARSET_PARAMETER);
+                if (charset != null) {
+                    try {
+                        return Charset.forName(charset);
+                    } catch (Exception e) {
+                        // unsupported charset, default to UTF-8
+                        return Charset.defaultCharset();
```

Comment:
change to utf-8

Reply:
https://github.com/micronaut-projects/micronaut-core/pull/11670/commits/43b3a2dfcfe58982a8ad3a6cb8004079b73dbf4a